public/app/features/correlations/Forms/TransformationsEditor.tsx

```diff
@@ -0,0 +1,281 @@
+import React, { useState } from 'react';
+import { useFormContext } from 'react-hook-form';
+
+import { GrafanaTheme2, SupportedTransformationType } from '@grafana/data';
+import { Stack } from '@grafana/experimental';
+import {
+  Button,
+  Field,
+  FieldArray,
+  Icon,
+  IconButton,
+  Input,
+  InputControl,
+  Label,
+  Select,
+  Tooltip,
+  useStyles2,
+} from '@grafana/ui';
+
+type Props = { readOnly: boolean };
+
+const getStyles = (theme: GrafanaTheme2) => ({
+  heading: css`
+    font-size: ${theme.typography.h5.fontSize};
+    font-weight: ${theme.typography.fontWeightRegular};
+  `,
+  // set fixed position from the top instead of centring as the container
+  // may get bigger when the for is invalid
+  removeButton: css`
+    margin-top: 25px;
+  `,
+});
+
+export const TransformationsEditor = (props: Props) => {
+  const { control, formState, register, setValue, watch, getValues } = useFormContext();
+  const { readOnly } = props;
+  const [keptVals, setKeptVals] = useState<{ [key: string]: { expression: string; mapValue: string } }>({});
+
+  const styles = useStyles2(getStyles);
+
+  const transformOptions = getTransformOptions();
+
+  return (
+    <FieldArray name="config.transformations" control={control}>
+      {({ fields, append, remove }) => (
+        <>
+          <Stack direction="column" alignItems="flex-start">
+            <div className={styles.heading}>Transformations</div>
+            {fields.length === 0 && <div> No transformations defined.</div>}
+            {fields.length > 0 && (
+              <div>
+                {fields.map((field, index) => {
+                  return (
+                    <Stack direction="row" key={field.id} alignItems="top">
+                      <Field
+                        label={
+                          <Stack gap={0.5}>
+                            <Label>Type</Label>
+                            <Tooltip
+                              content={
+                                <div>
+                                  <p>The type of transformation that will be applied to the source data.</p>
+                                </div>
+                              }
+                            >
+                              <Icon name="info-circle" size="sm" />
+                            </Tooltip>
+                          </Stack>
+                        }
+                        invalid={!!formState.errors?.config?.transformations?.[index]?.type}
+                        error={formState.errors?.config?.transformations?.[index]?.type?.message}
+                        validationMessageHorizontalOverflow={true}
+                      >
+                        <InputControl
+                          render={({ field: { onChange, ref, ...field } }) => (
+                            <Select
+                              {...field}
+                              onChange={(value) => {
+                                if (!readOnly) {
```

Comment:
I wanted the select list to open even if it is readonly so users can see their options, but not be allowed to change.

Reply:
This looks very confusing in read-only mode. Not a blocker for now but I think we should run it by UX. I'll also mention it in https://github.com/grafana/grafana/issues/65517. I think making the description accessible in read-only mode was suggested by @radiorental, but is it the right way of doing it?